cfgov/v1/models/base.py

```diff
@@ -186,7 +186,16 @@ def specific_categories_query(block, parent_slug):
                         search_query
                     ).distinct().exclude(id=self.id).order_by(
                         '-date_published'
-                    )[:block.value['limit']])
+                    ))
+                # Apply similar logic as snippets.py's filter_by_tags method
+                # to enable AND filtering
+                if block.value['and_filtering']:
+                    for tag in self.tags.names() or []:
+                        related[search_type_name] = (
+                            related[search_type_name].filter(tags__name=tag)
+                        )
+                related[search_type_name] = \
+                    related[search_type_name][:block.value['limit']]
```

Comment:
It's strange that the `limit` here is a `CharBlock` and not something like an [`IntegerBlock`](http://docs.wagtail.io/en/v1.8.1/topics/streamfield.html#integerblock). This shouldn't validate, but it does:

![image](https://cloud.githubusercontent.com/assets/654645/25281688/16738042-267c-11e7-9560-410d053e4323.png)

If you do this and try to preview or view the page, it 500s because `[:block.value['limit']]` doesn't work with non-integer text (the reason it works with integer text at all is somewhat surprising as well).

While you're messing with this block, how do you feel about fixing that?

(Also, minor style thing. To make the code a bit more readable, you could use a temporary variable for the query and only assign to `related[search_type_name]` at the end.)